src/configure/helper/graphHelper.ts

```diff
@@ -0,0 +1,212 @@
+                    {
+                        "startDate": startDate,
+                        "endDate": new Date(startDate.getFullYear() + 1, startDate.getMonth()),
+                        "value": secret
+                    }
+                ]
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+        .then((data) => {
+            return <AadApplication>{
+                appId: data.appId,
+                secret: secret
+            };
+        });
+    }
+
+    private static createSpn(credentials: TokenCredentials, appId: string, tenantId: string, retries: number = 0): Promise<any> {
+        let restClient = new RestClient(credentials);
+
+        return restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: `https://graph.windows.net/${tenantId}/servicePrincipals`,
+            queryParameters: {
+                "api-version": "1.6"
+            },
+            headers: {
+                "Content-Type": "application/json",
+            },
+            method: "POST",
+            body: {
+                "appId": appId,
+                "accountEnabled": "true"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+        .catch((error) => {
+            if(retries++ < this.retryCount) {
+                return Q.delay(this.retryTimeout)
+                .then(() => this.createSpn(credentials, appId, tenantId, retries));
+            }
+            throw error;
+        });
+    }
+
+    private static async createRoleAssignment(credentials: ServiceClientCredentials, scope: string, objectId: string, retries: number = 0): Promise<any> {
+        let restClient = new RestClient(credentials);
+        let roleDefinitionId = `${scope}/providers/Microsoft.Authorization/roleDefinitions/${this.contributorRoleId}`;
+        let guid = uuid();
+
+        return restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: `https://management.azure.com/${scope}/providers/Microsoft.Authorization/roleAssignments/${guid}`,
+            queryParameters: {
+                "api-version": "2015-07-01"
+            },
+            headers: {
+                "Content-Type": "application/json",
+            },
+            method: "PUT",
+            body: {
+                "properties": {
+                    "roleDefinitionId": roleDefinitionId,
+                    "principalId": objectId
+                }
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+        .catch((error) => {
+            if(retries++ < this.retryCount) {
+                return Q.delay(this.retryTimeout)
+                .then(() => this.createRoleAssignment(credentials, scope, objectId, retries));
+            }
+            throw error;
+        });
+    }
+
+    public static async getRefreshToken(session: AzureSession): Promise<Token> {
```

Comment:
should we arrange the functions in order of there calling pattern

Reply:
sure. But keeping the public methods first followed by private in order of call.